api-reference/headless-portal.mdx

```diff
@@ -1,15 +1,54 @@
 ---
-title: 'Headless Portal'
+title: 'Headless Support Portal'
 description: 'Give your customers visibility over their support requests.'
 ---
 
-Customer support portals empower your customers to view, create and reply to support requests as well as share visibility of issues with their team.
+<Warning>
+Headless Portal support is only available in our custom pricing tier. If you are interested in using this feature please reach out to use on sales@plain.com.
 
-Pre-built support portals can be hard to customize or theme and have to be separate from your product. Instead Plain provides a set of powerful APIs as well as an [example repository](https://github.com/team-plain/plain-headless-portal) which allows you to easily create a fully custom support portal.
+When building a headless support portal, you will have access to our engineering team, who can advise and help you. These docs are primarily intended to give you a high-level overview of how the process works. They do not aim to be exhaustive or self-serve.
 
-In order to use the Plain API you will need an API key, see [Getting Started](https://github.com/team-plain/plain-headless-portal?tab=readme-ov-file#getting-started) for full instructions. This API key should be kept secret since it will allow access to data from your whole Plain workspace. Calls to Plain's API **should be made from a server** rather than the client.
+</Warning>
 
-## Fetching threads
+Customer support portals allow your customers to view, create, and reply to support requests directly from your product.
+
+Plain provides a set of powerful APIs to do this, and you build the UI.
+
+This means you can build a support portal tailored to your product, look 100% white-label, and not require separate login credentials from your product.
+
+### Security & architecture
+
+To build a support portal, you will make API calls to [Plain's GraphQL API using an API key](/api-reference/graphql/authentication). This API key needs fairly broad permissions to be able to read threads and customer details as well as perform other actions such as send emails.
+
+For this reason it's **very very important to never make API calls to Plain directly from the client**. Doing so means you will be leaking the API key which is a big security risk.
+
+You must always make Plain API calls from an API you control. This allows you to implement access controls so that for example, a customer can only access their support requests etc.
+
+If you have leaked an API key by mistake, immediately delete it from your workspace settings and don't hesitate to reach out to us if we can help with any mitigation/investigations.
+
+### How does it work
+
+Details can vary depending on the experience you'd like to offer but in essence to build a portal you have 4 separate piece of work:
+
+1. Creating a tenant for each of your customers in Plain
```

Comment:
I think we need some terminology here to seperate customer as a group i.e. Acme vs customer as an individual working for Acme. There's a mix of this in this document and it is quite confusing

Reply:
Have added an intro to point to our data model docs!